server/index.js

```diff
@@ -33,6 +33,22 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:id', (req, res, next) => {
+  const parameterizedArray = [req.params.id];
+  const sql = `
+    select *
+      from "products"
+      where "productId" = $1;`;
+  db.query(sql, parameterizedArray)
+    .then(results => {
+      if (results.rows.length === 0) {
+        next(new ClientError(`cannot find id of ${req.params.id}`, 404));
+      }
+      res.json(results.rows[0]);
```

Comment:
Be sure to either `return` in your `if` block or add an `else` block. `next()` does not terminate the function, so if you want to stop there, you need to do one of those.